http/src/main/resources/swagger/api-docs.yaml

```diff
@@ -422,7 +424,279 @@ paths:
+                $ref: "#/components/schemas/ErrorReport"
+        "500":
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+    post:
+      summary: Creates a new persistent disk in the given project with the given name.
+      description: >
+        The request is completed without waiting for the persistent disk to be created in Google.
+        The disk status can be polled using the getDisk API.
+        Default labels diskName, googleProject, and creator cannot be overridden.
+      operationId: createDisk
+      tags:
+        - disks
+      parameters:
+        - in: path
+          name: googleProject
+          description: googleProject
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: name
+          description: diskName. only lowercase alphanumeric characters, numbers and dashes are allowed
+          required: true
+          schema:
+            type: string
+      requestBody:
+        $ref: "#/components/requestBodies/CreateDiskRequest"
+      responses:
+        "202":
+          description: Persistent disk creation request accepted
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+        "409":
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+        "500":
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+    delete:
+      summary: Deletes an existing persistent disk in the given project
+      description: Deletes a persistent disk
+      operationId: deleteDisk
+      tags:
+        - disks
+      parameters:
+        - in: path
+          name: googleProject
+          description: googleProject
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: name
+          description: diskName
+          required: true
+          schema:
+            type: string
+      responses:
+        "202":
+          description: Persistent disk deletion request accepted
+        "403":
+          description: User does not have permission to perform action on persistent disk
+        "404":
+          description: Persistent disk not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+        "500":
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+    patch:
+      summary: Updates the configuration of a persistent disk
```

Comment:
if the disk is being used by a running VM, will we be able to update the disk?

Reply:
You can resize a disk while it's attached to a running VM.

My PR involves adding validation/documentation about what operations are/aren't allowed with disks & VMs.